modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -87,15 +97,24 @@ def report msg, response, error_only
       raise Proxy::Dns::Error.new("Unknown error while processing '#{msg}'")
     end
 
-    def match_zone(fqdn)
+    def match_zone(fqdn_or_ip)
```

Comment:
Pass in zone_name as a parameter. You can remove conditional logic used to determine whether zone is forward or ptr one.